### PR TITLE
feature added templates/lanced as vectordb 

### PR DESCRIPTION
**Pull request by akashAD98**
added lanced as vectordb example in template

### CODE REVIEW

**Comment by akashAD98** [comment]
fix https://github.com/langchain-ai/langchain/pull/3411

**Review by efriis — CHANGES_REQUESTED** [review]
Hey @akashAD98 ! I tried debugging this but keep running into the following. Could you test and see if you get the same?

```Traceback (most recent call last):
  File "/Users/erickfriis/langchain/oss-py/templates/rag-lancedb/.venv/lib/python3.11/site-packages/uvicorn/protocols/http/h11_impl.py", line 408, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/erickfriis/langchain/oss-py/templates/rag-lancedb/.venv/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/erickfriis/langchain/oss-py/templates/rag-lancedb/.venv/lib/python3.11/site-packages/fastapi/applications.py", line 1106, in __call__
    await super().__call__(scope, receive, send)
  File "/Users/erickfriis/langchain/oss-py/templates/rag-lancedb/.venv/lib/python3.11/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/erickfriis/langchain/oss-py/templates/rag-lancedb/.venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/Users/erickfriis/langchain/oss-py/templates/rag-lancedb/.venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/Users/erickfriis/langchain/oss-py/templates/rag-lancedb/.venv/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/Users/erickfriis/langchain/oss-py/templates/rag-lancedb/.venv/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/Users/erickfriis/langchain/oss-py/templates/rag-lancedb/.venv/lib/python3.11/site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
    raise e
  File "/Users/erickfriis/langchain/oss-py/templates/rag-lancedb/.venv/lib/python3.11/site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
  File "/Users/erickfriis/langchain/oss-py/templates/rag-lancedb/.venv/lib/python3.11/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/Users/erickfriis/langchain/oss-py/templates/rag-lancedb/.venv/lib/python3.11/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/Users/erickfriis/langchain/oss-py/templates/rag-lancedb/.venv/lib/python3.11/site-packages/starlette/routing.py", line 69, in app
    await response(scope, receive, send)
  File "/Users/erickfriis/langchain/oss-py/templates/rag-lancedb/.venv/lib/python3.11/site-packages/sse_starlette/sse.py", line 233, in __call__
    async with anyio.create_task_group() as task_group:
  File "/Users/erickfriis/langchain/oss-py/templates/rag-lancedb/.venv/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 597, in __aexit__
    raise exceptions[0]
  File "/Users/erickfriis/langchain/oss-py/templates/rag-lancedb/.venv/lib/python3.11/site-packages/sse_starlette/sse.py", line 236, in wrap
    await func()
  File "/Users/erickfriis/langchain/oss-py/templates/rag-lancedb/.venv/lib/python3.11/site-packages/sse_starlette/sse.py", line 221, in stream_response
    async for data in self.body_iterator:
  File "/Users/erickfriis/langchain/oss-py/templates/rag-lancedb/.venv/lib/python3.11/site-packages/langserve/server.py", line 800, in _stream_log
    "data": well_known_lc_serializer.dumps(data),
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/erickfriis/langchain/oss-py/templates/rag-lancedb/.venv/lib/python3.11/site-packages/langserve/serialization.py", line 176, in dumps
    return json.dumps(obj, cls=_LangChainEncoder)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/erickfriis/.pyenv/versions/3.11.4/lib/python3.11/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "/Users/erickfriis/.pyenv/versions/3.11.4/lib/python3.11/json/encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/erickfriis/.pyenv/versions/3.11.4/lib/python3.11/json/encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "/Users/erickfriis/langchain/oss-py/templates/rag-lancedb/.venv/lib/python3.11/site-packages/langserve/serialization.py", line 98, in default
    return super().default(obj)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/erickfriis/.pyenv/versions/3.11.4/lib/python3.11/json/encoder.py", line 180, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type ndarray is not JSON serializable```

File: templates/rag-lancedb/README.md
Comment (efriis) [inline templates/rag-lancedb/README.md:1]: Also please fill this in!

**Comment by akashAD98** [comment]
@efriis  Currently, i don't have openai for testing but soon it test it.
where is the error?in chain.py?  any reference so ill check that

File: templates/rag-lancedb/README.md
Comment (akashAD98) [reply templates/rag-lancedb/README.md:1]: yes

**Comment by efriis** [comment]
Hey @akashAD98 ! Closing this til you set up OpenAI for testing. When you're able to get it working, feel free to reopen or open a new PR!

If you have access to a different LLM than OpenAI, you're welcome to change the template to use your llm of choice :)

**Comment by akashAD98** [comment]
@efriis  i have tested it on collab its working fine

image

**Pull request closed without merging**